### PR TITLE
[11.x.x] Add terrain shader tag

### PR DESCRIPTION
**Pull request by barringtonunity3d**
### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR ###
This PR adds a Terrain shader tag that's used to distinguish if a shader supports terrain features.
The tag will be used in conjunction with https://github.com/Unity-Technologies/TerrainTools/pull/595

---
### Testing status ###

Tested using 2021.1.0a4 branched off of Master (0e0baaa)
- [x] Create a terrain in scene
- [x] Sculpt the terrain
- [x] Paint the terrain
- [x] Add details to the terrain
- [x] Test Material Override Detection

---
### Comments to reviewers
Please test with https://github.com/Unity-Technologies/TerrainTools/pull/595

### CODE REVIEW

File: com.unity.render-pipelines.universal/Shaders/Terrain/TerrainLit.shader [hunk 1]
```diff
@@ -48,7 +48,7 @@ Shader "Universal Render Pipeline/Terrain/Lit"
 
     SubShader
     {
-        Tags { "Queue" = "Geometry-100" "RenderType" = "Opaque" "RenderPipeline" = "UniversalPipeline" "UniversalMaterialType" = "Lit" "IgnoreProjector" = "False"}
+        Tags { "Queue" = "Geometry-100" "RenderType" = "Opaque" "RenderPipeline" = "UniversalPipeline" "UniversalMaterialType" = "Lit" "IgnoreProjector" = "False" "Terrain" = "true"}
```
Comment (zeroyao) [inline com.unity.render-pipelines.universal/Shaders/Terrain/TerrainLit.shader:51]: Maybe use a more specific tag name? How about "WorkForTerrain"?

**Review by zeroyao — APPROVED** [review]
Tag name could be more specific, but "Terrain" should work too!

**Comment by phi-lira** [comment]
Is there a documentation added for this new terrain tag? Is the new Terrain tag a requirement for all terrain shader? What's the expected behaviour for a shader that doesn't declare this tag? 

Should we update the TerrainAdd + TerrainBase shaders? https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.universal/Shaders/Terrain/TerrainLitAdd.shader

**Comment by barringtonunity3d** [comment]
> Is there a documentation added for this new terrain tag? Is the new Terrain tag a requirement for all terrain shader? What's the expected behaviour for a shader that doesn't declare this tag?
> 
> Should we update the TerrainAdd + TerrainBase shaders? https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.universal/Shaders/Terrain/TerrainLitAdd.shader

There currently isn't any documentation for the new tag. It's not a requirement for the shaders as it's only used to detect whether a user is selecting a terrain shaders to use when creating terrains from the Terrain Toolbox. If they aren't using the Terrain Shaders we provide they're shown a warning,

Material_Override_Instance_Fix

I don't believe it's necessary to update the Base and Add shaders, but let me follow up with Yao.

File: com.unity.render-pipelines.universal/Shaders/Terrain/TerrainLit.shader (hunk 1 above)
Comment (cdxntchou) [reply com.unity.render-pipelines.universal/Shaders/Terrain/TerrainLit.shader:51]: "TerrainCompatible" ?

**Review by sebastienlagarde — APPROVED** [review]

**Review by LandonTownsendUnity — CHANGES_REQUESTED** [review]
Needs changelog

**Comment by sebastienlagarde** [comment]
Ping: any update on this PR? @phi-lira , @cdxntchou , @LandonTownsendUnity  ?

**Comment by sebastienlagarde** [comment]
Double ping :) 
Ping: any update on this PR? @phi-lira , @cdxntchou , @LandonTownsendUnity ?

**Comment by barringtonunity3d** [comment]
Hey @sebastienlagarde,

Just wanted to give you an update from my end

@LandonTownsendUnity is currently out on vacation this week. He had a few higher priority QA related tasks to complete before finishing this one. He should be back next week.

I'll ping @cdxntchou to see if he has time to review.

@phi-lira recommended adding documentation so I reached out to our Technical Writers who made a ticket to get this done https://unity3d.atlassian.net/browse/DOC-2524. I just pinged him to see how much progress has been made.

**Review by cdxntchou — APPROVED** [review]

**Review by phi-lira — APPROVED** [review]
LGTM.

**Comment by phi-lira** [comment]
Resolved conflicts in changelog and runned ABV.

File: com.unity.render-pipelines.high-definition/CHANGELOG.md [hunk 2]
```diff
@@ -66,6 +66,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Added missing documentation for unsupported SG RT nodes and light's include for raytracing attrbute.
 - Added documentation for LODs not being supported by ray tracing.
 - Added more options to control how the component of motion vectors coming from the camera transform will affect the motion blur with new clamping modes.
+- Added a terrain compatiblity shader tag for validating whether user overridden materials support terrain rendering functionalities.
```
Comment (oleks-k) [inline com.unity.render-pipelines.high-definition/CHANGELOG.md:69]: Suggestion:
"Before this version, when using Terrain Toolbox, if a user selected a Material that did not use one of Unity's pre-built Terrain shaders in the Material Override field, Unity showed a warning. This version of the package implements the TerrainCompatible SubShader Tag. Adding this tag to a custom Terrain shader tells Unity that the shader is compatible with the Terrain system, and Unity does not show the warning if a user selects a Material with such shader in the Material Override field."

File: com.unity.render-pipelines.universal/CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Added
 - Added a supported MSAA samples count check, so the actual supported MSAA samples count value can be assigned to RenderTexture descriptors.
+- Added a terrain compatiblity shader tag for validating whether user overridden materials support terrain rendering functionalities.
```
Comment (oleks-k) [inline com.unity.render-pipelines.universal/CHANGELOG.md:11]: See the comment in HDRP (if that comment is correct, use it here as well).

File: com.unity.render-pipelines.high-definition/CHANGELOG.md (hunk 2 above)
Comment (barringtonunity3d) [reply com.unity.render-pipelines.high-definition/CHANGELOG.md:69]: I made a few edits

"Before this version, when ~~using Terrain Toolbox, if~~  a user selected a Material that did not use one of Unity's pre-built Terrain shaders in **a** ~~the~~ Material Override field, Unity **did not** show~~ed~~ a warning. This version of the package implements the TerrainCompatible SubShader Tag. Adding this tag to a custom Terrain shader tells Unity that the shader is compatible with the Terrain system, and Unity does not show the warning if a user selects a Material with such shader in **a** ~~the~~ Material Override field."

Reasoning for changes
1. These changes will be used with and without Terrain Toolbox (Removed "using Terrain Toolbox")
2. There is a couple of fields where the user can override/change the material (Changed "the" -> "a")
3. Before this change there wasn't a warning for users (Added "did not")

Comment (oleks-k) [reply com.unity.render-pipelines.high-definition/CHANGELOG.md:69]: Here's a shorter version:
- Added the TerrainCompatible SubShader Tag. Use this Tag in your custom shader to tell Unity that the shader is compatible with the Terrain system.

File: com.unity.render-pipelines.universal/CHANGELOG.md (hunk 3 above)
Comment (barringtonunity3d) [reply com.unity.render-pipelines.universal/CHANGELOG.md]: Changed to match revisions made to URP's ChangeLog

**Review by oleks-k — APPROVED** [review]
Approved with my comment. 
I wrote down the notes in the ticket:
https://unity3d.atlassian.net/browse/DOC-2524

**Review by GrantLamb-Unity — APPROVED** [review]

**Comment by sebastienlagarde** [comment]
So all valid. Merging. Note: does it need to be backport on 10.x?

**Comment by sebastienlagarde** [comment]
Landon have remove his review but the request is still there ( LandonTownsendUnity removed their request for review 17 hours ago), so all reviewer are green.

**Comment by sebastienlagarde** [comment]
> So all valid. Merging. Note: does it need to be backport on 10.x?

Sorry it was in the title :)

**Comment by Dawie3565** [comment]
Hello 
has tag TerrainCompatible = true been backported for HDRP 10x yet?

HDRP api 10x i am still getting "Can't use materials with shaders which need tangent geometry on terrain, use shaders in Nature/Terrain instead

i am looking for way to suppress this in api 10.8.1 and higher 
i only found a change log for 11x

**Pull request merged**